### ISSUE TITLE
Roslyn suggests to use C#7 language construct when they aren't valid

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#18983]
**Version Used**: 
VS Version 15.1 (26403.7) Release

**Steps to Reproduce**:

1. Create new console application
2. Add nuget package Microsoft.Net.Compilers which version doesn't support C#7 e.g.
```Install-Package Microsoft.Net.Compilers -Version 1.3.2 ```
3. Add some code which can be simplified with C#7 language e.g.
```cs
      int x;
      int.TryParse("0", out x);
```

**Expected Behavior**:
Roslyn doesn't suggest to inline 'int x' into C#7 out int x

**Actual Behavior**:
Roslyn suggests to convert code into the following:
```cs
      int.TryParse("0", out int x);
```
After the change the application doesn't compile. Roslyn also doesn't warn about the error so you can turn a lot of code into C#7 constructs before noticing that it doesn't compile anymore.

**Comment by Pilchie** [comment]
The IDE is unfortunately unaware of the NuGet package, and is based only on the value of the "/langversion" switch.  The default value for that switch is "default", which is the latest major version.  In the IDE this is 7.0, which is why these features get offered. 

You can set the language version to 6 in your project properties to get the IDE to respect that.

(And I think @jasonmalinowski had some thoughts about fixing this in the future so the IDE could find out the maximum language version supported by the compiler in use).

**Comment by Pilchie** [comment]
@jasonmalinowski did you have another issue already tracking that?

**Comment by jasonmalinowski** [comment]
I guess I had asked if @jcouv had something tracking it, but didn't follow up. @jcouv?

**Comment by jcouv** [comment]
Forwarded you notes from last discussion on this (how langver interacts with compiler and analyzers). Key points:
> We discussed the idea of providing analyzer-writers a way to check the maximum lang version of nodes in a tree. We will wait for user feedback before exploring this further.
> We discussed the problem of having the IDE make suggestions that are not appropriate for an older version of the language. But we decided to keep this behavior for now.

**Comment by Pilchie** [comment]
Internal discussion about whether we could do this for the NuGet package/`CscToolPath` case is continuing.

**Comment by tannergooding** [comment]
For C# 7.1, where you have to opt-in to the new language features, it can get confusing when the IDE suggests a feature or when the IDE says everything is clean, but build (F6) fails because the PackageRef compiler doesn't recognize the code as valid.

**Comment by CyrusNajmabadi** [comment]
Closing out as by design from the perspective of the feature.  IF hte langversion is 7 we respect htat.  if you set it to 6, we will respect that as well.
